### PR TITLE
Fixes #71 - Integrate with Permissions

### LINKED ISSUE #71: Permissions API integration needs to be improved
See these issues:
- https://github.com/w3c/permissions/issues/74
- https://github.com/w3c/permissions/issues/33

### PR DESCRIPTION
**Pull request by miketaylr**
In theory this is sufficient, but I don't know a ton about Notifications.

cc @marcoscaceres

***
<a href="https://whatpr.org/notifications/172.html" title="Last updated on Nov 15, 2021, 4:45 PM UTC (147083e)">Preview</a> | <a href="https://whatpr.org/notifications/172/e0cadb2...147083e.html" title="Last updated on Nov 15, 2021, 4:45 PM UTC (147083e)">Diff</a>

### CODE REVIEW

**Comment by miketaylr** [comment]
> Some checks were not successful
> Trailing whitespace:
324:This specification's "<code>default</code>" <a>permission</a> state maps to the 

I feel set up.

File: notifications.bs [hunk 1]
```diff
@@ -729,8 +735,20 @@ constructor steps are:
 return the <a>permission</a> for the <a>current settings object</a>'s
 <a for="environment settings object">origin</a>.
 
-<p class=note>If you edit standards please refrain from copying the above. Synchronous
-permissions are like synchronous IO, a bad idea.
+<div class=note>
+ <p>If you edit standards please refrain from copying the above. Synchronous permissions are like
+ synchronous IO, a bad idea.
+
+ <p>Developers are encouraged to use the [[Permissions]] API's {{Permissions/query()}} method
```
Comment (annevk) [inline notifications.bs:742]: We typically write the name of the standard in full in the sentence and put the reference at the end of the paragraph, after the dot.

File: notifications.bs [hunk 2]
```diff
@@ -729,8 +735,20 @@ constructor steps are:
...
+ instead.
+
+ <pre class="example" id="permissions-query-example">
+    const permission = await navigator.permissions.query({name: "notifications"});
+    if (permission.state === "granted") {
+       // We have permission to use the API...
```
Comment (annevk) [inline notifications.bs:748]:
```suggestion
       // We have permission to use the API…
```

File: notifications.bs [hunk 3]
```diff
@@ -739,35 +757,38 @@ method steps are:
...
 
   <ol>
-   <li><p>Let <var>permission</var> be <a>permission</a> for <var>origin</var>.
+   <li><p>Run the <a for="powerful feature">permission query algorithm</a> with
+   <var>permissionDescriptor</var> and <var>permissionStatus</var>.
 
-   <li><p>If <var>permission</var> is "<code>default</code>", then ask the user whether showing
-   notifications for <var>origin</var> is acceptable. If it is, then set <var>permission</var> to
-   "<code>granted</code>"; otherwise "<code>denied</code>".
+   <li><p>Let <var>permission state</var> be <var>permissionStatus</var>'s
```
Comment (annevk) [inline notifications.bs:776]: Can we rename this to _permissionState_ throughout for consistency?

File: notifications.bs [hunk 4]
```diff
@@ -739,35 +757,38 @@ method steps are:
...
+   {{PermissionStatus/state}}.
+
+   <li><p>If <var>permission state</var> is {{PermissionState/"prompt"}}, then override
+   <var>permission state</var> with "<code>default</code>".
```
Comment (annevk) [inline notifications.bs:780]: override with -> set to

File: notifications.bs [hunk 5]
```diff
@@ -317,6 +317,12 @@ meaning "<code>granted</code>". In that case
 "<code>granted</code>" is simply returned as there would be no reason
 for the application to ask for <a>permission</a>.
 
+<h3 id="permissions-integration">Permissions integration</h3>
+
+<p>The Notifications API is a <a>powerful feature</a>. [[!Permissions]]
```
Comment (annevk) [inline notifications.bs:322]: I saw some of the back-and-forth about this sentence. Isn't a problem here that #93 is still open?

**Review by annevk — COMMENTED** [review]
Sorry for the slow turnaround. Will prioritize this a bit more. Looking fairly solid though, just one somewhat substantive comment.

File: notifications.bs (hunk 3 above)
Comment (miketaylr) [reply notifications.bs:776]: Sure thing.

File: notifications.bs (hunk 5 above)
Comment (miketaylr) [reply notifications.bs:322]: I guess there's 2 ways to go about this: 

1) land as-is (w/o "allowed in non-secure contexts" flag set), and do the work to close #93.
2) land with the "allowed in non-secure contexts" flag set, do the work to close #93, and when remove the flag at the same time.

I'm not sure I see any real advantage to either approach (but we should do the work to close #93, yes), the spec today is already not matching what WebKit, Gecko, and Chromium do wrt secure context requirements.

WDYT @annevk?

File: notifications.bs [hunk 6]
```diff
@@ -317,6 +317,14 @@ meaning "<code>granted</code>". In that case
 "<code>granted</code>" is simply returned as there would be no reason
 for the application to ask for <a>permission</a>.
 
+<h3 id="permissions-integration">Permissions integration</h3>
+
+<p>The Notifications API is a <a>powerful feature</a> which is identified by the string
+"notifications" and has its <a for="powerful feature">allowed in
+non-secure contexts</a> flag set. [[!Permissions]]
```
Comment (marcoscaceres) [inline notifications.bs:324]: Should just be: 

```suggestion
<p>The Notifications API is a <a>powerful feature</a> that is identified by the string
"notifications". [[!Permissions]]
```

File: notifications.bs (hunk 5 above)
Comment (marcoscaceres) [reply notifications.bs]: Suggested we drop the flag, per 1.

File: notifications.bs (hunk 6 above)
Comment (annevk) [reply notifications.bs:324]: No newlines inside phrasing elements.

**Review by annevk — COMMENTED** [review]
I guess I don't have a strong opinion on whether we keep the flag. I'd lean toward keeping it as the specification remains self-consistent in that case.

**Comment by miketaylr** [comment]
> I'd lean toward keeping it as the specification remains self-consistent in that case.

Makes sense to me. Let's land it w/ the flag, and I'll try to get to #93 in the next week and remove it as part of that PR.

File: notifications.bs [hunk 7]
```diff
@@ -317,6 +317,14 @@ meaning "<code>granted</code>". In that case
...
 for the application to ask for <a>permission</a>.
 
+<h3 id="permissions-integration">Permissions integration</h3>
+
+<p>The Notifications API is a <a>powerful feature</a> which is identified by the string
+"notifications" and has its <a for="powerful feature">allowed in non-secure contexts</a> flag set.
+[[!Permissions]]
+
+<p class="note">This specification's "<code>default</code>" <a>permission</a>
+state maps to the [=permission/prompt=] permission state in the Permissions API.
```
Comment (annevk) [inline notifications.bs:328]:
```suggestion


```

**Review by annevk — APPROVED** [review]

**Pull request merged**